src/activity_actor.cpp

```diff
@@ -8736,61 +8763,83 @@ void pulp_activity_actor::send_final_message( Character &you ) const
-            skill += " and your body is not fit";
-        }
-    } else if( you.get_skill_level( skill_swimming ) < 2 ) {
-        skill = "even if your body is not fit";
-    } else {
-        skill = "";
+    if( you.get_skill_level( skill_swimming ) < 2 ) {
+        skill = _( "Heavy breathing, you think your form is far from perfect for such activities." );
     }
 
-    std::string pry = ".";
+    std::string pry;
     if( pd.pry_tool == pd.bash_tool ) {
-        // use default assignment
+        // bail out
     } else if( pd.used_pry ) {
-        pry = string_format( ".  You also were lucky to carry %, it helped a lot with more armoured bodies.",
-                             pd.pry_tool );
+        pry = string_format(
+                  _( "You praise yourself for bringing %1$s, it helped a lot with more armoured bits." ),
+                  pd.pry_tool );
     } else if( pd.couldnt_use_pry ) {
-        pry = string_format( ".  You wish you had anything like a crowbar, you had a really hard time without it." );
+        pry = _( "You also consider finding something like a crowbar, you had a really hard time without it." );
     }
 
-    you.add_msg_player_or_npc(
-        //~ %1$s is message about tools used in pulping, %2$s is a note if your skill is too low (optional, can be empty),
-        //~ 3$s is message about you using/not being able to use prying tool (optional, can be `.`).
-        //~ see pulp_activity_actor::send_final_message() for full context
-        n_gettext( "You finished pulping the corpse%1$s, %2$s%3$s",
-                   "You finished pulping the corpses%1$s, %2$s%3$s", unpulped_corpses_qty ),
-        n_gettext( "<npcname> finished pulping the corpse.",
-                   "<npcname> finished pulping the corpses.", unpulped_corpses_qty ),
-        tools, skill, pry
-    );
+    std::string final_message;
+
+    final_message += msg;
+    final_message += tools.empty() ? "" : "  " + tools;
+    final_message += skill.empty() ? "" : "  " + skill;
+    final_message += pry.empty() ? "" : "  " + pry;
+
+    you.add_msg_if_player( final_message );
 
     if( way_too_long_to_pulp ) {
-        you.add_msg_player_or_npc(
-            n_gettext( "You left one corpse unpulped, for you have no tool heavy enough to deal with it.",
-                       "You left some corpses unpulped, for you have no tool heavy enough to deal with it.",
-                       unpulped_corpses_qty ),
-            n_gettext( "<npcname> left one corpse unpulped, for %1$s has no tool heavy enough to deal with it.",
-                       "<npcname> left some corpses unpulped, for %1$s has no tool heavy enough to deal with it.",
-                       unpulped_corpses_qty ),
-            you.male ? pgettext( "npc", "he" ) : pgettext( "npc", "she" )
+        you.add_msg_if_player(
+            n_gettext( _( "You left one corpse unpulped, for you have no tool heavy enough to deal with it." ),
```

Comment:
You're double-translating here (which breaks pluralization), just pass these strings to n_gettext.